src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.CallSiteContainerRewriter.cs

```diff
@@ -377,6 +377,17 @@ public override SyntaxNode VisitClassDeclaration(ClassDeclarationSyntax node)
                     return node.WithMembers(newMembers);
                 }
 
+                public override SyntaxNode VisitRecordDeclaration(RecordDeclarationSyntax node)
+                {
+                    if (node != ContainerOfStatementsOrFieldToReplace)
+                    {
+                        return base.VisitRecordDeclaration(node);
+                    }
+
+                    var newMembers = VisitList(ReplaceMembers(node.Members, global: false));
+                    return node.WithMembers(newMembers);
+                }
+
                 public override SyntaxNode VisitStructDeclaration(StructDeclarationSyntax node)
```

Comment:
consider having all four of these methods calling into one common helper that operates on TypeDeclarationSyntax